urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/agent/infocollect/DynamicDiskIOInfo.java

```diff
@@ -0,0 +1,149 @@
+			br = new BufferedReader(new InputStreamReader(bis));
+			String lineStr;
+			while ((lineStr = br.readLine()) != null) {
+				lineStr = lineStr.trim().replaceAll(" {2,}", " ");
+				String[] array = lineStr.split(" ");
+				if (diskNameList.contains(array[0])) {
+					ioTs += Double.parseDouble(array[1]);
+					readSpeed += Double.parseDouble(array[2]);
+					writeSpeed += Double.parseDouble(array[3]);
+					readKB += Double.parseDouble(array[4]);
+					writeKB += Double.parseDouble(array[5]);
+				}
+			}
+			readKB /= 1024;
+			writeKB /= 1024;
+			try {
+				if (p.waitFor() != 0) {
+					if (p.exitValue() == 1)
+						LOG.error("Command execution failed!");
+				}
+			} catch (InterruptedException e) {
+				// TODO Auto-generated catch block
+				LOG.error(LogUtils.getTrace(e));
+			}
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			LOG.error(LogUtils.getTrace(e));
+		} finally {
+			try {
+				if (br != null)
+					br.close();
+				if (bis != null)
+					bis.close();
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				LOG.error(LogUtils.getTrace(e));
+			}
+		}
+		SharedArea.dynamicInfoSerializable.setIoTs((int) ioTs);
+		SharedArea.dynamicInfoSerializable.setReadKB((int) readKB);
+		SharedArea.dynamicInfoSerializable.setReadSpeed((int) readSpeed);
+		SharedArea.dynamicInfoSerializable.setWriteKB((int) writeKB);
+		SharedArea.dynamicInfoSerializable.setWriteSpeed((int) writeSpeed);
+	}
+
+	private List<String> getDiskNameByFile() {
+		List<String> diskNameList = new ArrayList<String>();
+		BufferedReader br = null;
+		try {
+			br = new BufferedReader(new FileReader("/proc/partitions"));
+			String line = null;
+			try {
+				while ((line = br.readLine()) != null) {
+					if (line.length() != 0) {
+						line = line.trim().replaceAll(" {2,}", " ");
+						String[] array = line.split(" ");
+						if (isNumeric(array[0])) {
+							int i = Integer.parseInt(array[0]);
+							int j = Integer.parseInt(array[1]);
+							if (i % 8 == 0 && j % 16 == 0) {
+								diskNameList.add(array[3]);
+							}
+						}
+					}
+				}
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				LOG.error(LogUtils.getTrace(e));
+			}
+		} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			LOG.error(LogUtils.getTrace(e));
+		} finally {
+			try {
+				if (br != null)
+					br.close();
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				LOG.error(LogUtils.getTrace(e));
+			}
+		}
+		return diskNameList;
+	}
+
+	private boolean isNumeric(String str) {
+		for (int i = str.length(); --i >= 0;) {
+			if (!Character.isDigit(str.charAt(i))) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	@Override
+	public void run() {
+		// To change body of implemented methods use File | Settings | File
+		// Templates.
+		while (running) {
+			getDiskIOInfoByCmd();
+			try {
+				Thread.sleep(samplingInterval);
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)